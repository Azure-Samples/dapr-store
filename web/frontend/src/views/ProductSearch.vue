<!--
// ----------------------------------------------------------------------------
// Copyright (c) Ben Coleman, 2020
// Licensed under the MIT License.
//
// Dapr Store frontend - Show product results from a search
// ----------------------------------------------------------------------------
-->

<template>
  <div>
    <h1>Search results for: <i>{{ this.$route.params.query }}</i></h1>
    <error-box :error="error" />
    <product-list v-if="!error" :products="products" />
  </div>
</template>

<script>
import api from '../services/api'
import ProductList from '../components/ProductList'
import ErrorBox from '../components/ErrorBox'

export default {
  name: 'ProductSearch',

  components: {
    'product-list': ProductList,
    'error-box': ErrorBox
  },

  data() {
    return {
      products: null,
      error: null
    }
  },

  watch: {
    async $route() {
      this.doSearch()
    }
  },

  async mounted() {
    this.doSearch()
  },

  methods: {
    async doSearch() {
      try {
        this.products = null
        this.products = await api.productSearch(this.$route.params.query)
      } catch (err) {
        this.error = err
      }
    }
  }
}
</script>
