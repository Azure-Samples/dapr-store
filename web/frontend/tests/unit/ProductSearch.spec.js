import { createLocalVue, mount } from '@vue/test-utils'
import flushPromises from 'flush-promises'

import BootstrapVue from 'bootstrap-vue'
const localVue = createLocalVue()
localVue.use(BootstrapVue)

import VueRouter from 'vue-router'
localVue.use(VueRouter)
const router = new VueRouter()

import ProductSearch from '@/views/ProductSearch.vue'

jest.mock('@/services/api')

describe('ProductSearch.vue', () => {
  it('renders search for Ascot', async () => {
    router.push({ name: 'search', params: { query: 'Ascot' } })

    const wrapper = mount(ProductSearch, {
      localVue,
      router
    })

    await flushPromises()
    expect(wrapper).toMatchSnapshot()
  })
})