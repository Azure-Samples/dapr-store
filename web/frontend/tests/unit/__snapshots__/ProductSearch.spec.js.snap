// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductSearch.vue renders search for Ascot 1`] = `
<div>
  <h1>Search results for: <i>Ascot</i></h1>
  <!---->
  <div>
    <!---->
    <div class="card">
      <!---->
      <!---->
      <div class="card-body">
        <!---->
        <!---->
        <div class="row">
          <div class="col"><a href="#/product/prd3" class="" target="_self">
              <h4 class="card-title">
                Paisley Cravat Ascot Tie
              </h4>
            </a>
            <p class="card-text">
              Black &amp; Red Paisley Cravat
              <br><br>
            <h4>Â£11.2</h4>
            </p> <a role="button" aria-disabled="true" tabindex="-1" href="#" target="_self" class="btn d-none d-md-inline disabled btn-primary disabled">
              <fa icon="shopping-cart"></fa>
              &nbsp; Add to Cart
            </a>
          </div>
          <div class="flex-grow-0 d-none d-md-block col">
            <div class="product-img">
              <!----> <a href="#/product/prd3" class="" target="_self"><img src="/img/catalog/3.jpg"></a></div>
          </div>
        </div>
      </div>
      <!---->
      <!---->
    </div>
  </div>
</div>
`;
